### PR TITLE
Enabled APM with socket communication by default

### PR DESCRIPTION
**Pull request by clamoriniere**
#### What this PR does / why we need it:

Enable by default the "trace-agent" with socket communication.

add 2 new settings  for APM:
* `datadog.apm.portEnabled` use to enable the TCP communication (false by default, to avoid the hostPort mount by default)
* `datadog.apm.socketEnabled` use to enable the trace-agent with only socket communication support. (`true` by default)

To maintain the backward compatibility:
* we keep `datadog.apm.enabled` with `false` as the default value. If a user set it to `true` the behavior doesn't change: the trace-agent is enabled and the hostPort is open. But this parameter is now flagged as `deprecated` in favour of `datadog.apm.portEnabled`
* `datadog.apm.useSocketVolume` is now deprecated in favour of `datadog.apm.socketEnabled`.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has beed updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

File: charts/datadog/README.md [hunk 1]
```diff
@@ -140,6 +140,28 @@ helm upgrade -f datadog-values.yaml <RELEASE_NAME> datadog/datadog --recreate-po
 
 See the [All configuration options](#all-configuration-options) section to discover all possibilities offered by the Datadog chart.
 
+### Enabling APM and Trace
+
+APM is enabled by default thrue socket communication; more details about the applications configuration are available on the [official documentation](https://docs.datadoghq.com/agent/kubernetes/apm/?tab=helm)
```
Comment (vboulineau) [inline charts/datadog/README.md:145]: s/thrue/through/

File: charts/datadog/templates/NOTES.txt [hunk 2]
```diff
@@ -126,6 +126,27 @@ Trace Agent liveness probe port ({{ $liveness.port }}) is different from the con
...
 {{- end }}
 
+{{- if .Values.datadog.apm.enabled }}
+
+#################################################################
+####               WARNING: Deprecation notice               ####
+#################################################################
+
+The option `datadog.apm.enabled` is deprecated, please use `datadog.apm.portEnabled` to enable TCP communication to the trace-agent.
+Or `datadog.apm.socketEnabled` to enable unix or name-pipe communication.
```
Comment (vboulineau) [inline charts/datadog/templates/NOTES.txt:136]: Clarify it's activated by default

**Review by vboulineau — APPROVED** [review]
We should clarify that, with default setup, applications need to mount `hostPath` to be able to send traces.

File: charts/datadog/templates/agent-services.yaml [hunk 3]
```diff
@@ -81,7 +81,7 @@ spec:
       port: {{ .Values.datadog.dogstatsd.port }}
       targetPort: {{ .Values.datadog.dogstatsd.port }}
       name: dogstatsd
-{{- if .Values.datadog.apm.enabled }}
+{{- if or .Values.datadog.apm.enabled .Values.datadog.apm.portEnabled }}
```
Comment (clamoriniere) [inline charts/datadog/templates/agent-services.yaml:84]: this if is used in different places, maybe you can make a function to replace it

File: charts/datadog/templates/_daemonset-volumes-linux.yaml [hunk 4]
```diff
@@ -24,7 +24,7 @@
     type: File
   name: kubelet-ca
 {{- end }}
-{{- if .Values.datadog.apm.useSocketVolume }}
+{{- if or .Values.datadog.apm.useSocketVolume .Values.datadog.apm.socketEnabled }}
```
Comment (clamoriniere) [inline charts/datadog/templates/_daemonset-volumes-linux.yaml:27]: could you create a function for this `if` too?

**Review by clamoriniere — COMMENTED** [review]
/lgtm 
🏆

**Comment by caleb15** [comment]
Is there any advantage to using sockets over ports?

**Pull request merged**